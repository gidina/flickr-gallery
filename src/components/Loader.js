import React from 'react';
import "./Loader.css";

const Loader = ({color}) => {
    console.log("LOADER");
    return (
        <div className="loader" style={{ border: '10px solid #f3f3f3', borderTop: `10px solid ${color}` }}></div>
    );
}

// Loader.protoTypes = {
//     color: React.PropTypes.string.isRequired
// };

export default Loader;