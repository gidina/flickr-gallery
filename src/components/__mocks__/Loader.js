import React from "react";
export default class Loader extends React.Component {
    render() {
        return <loader-mock {...this.props}>{this.props.children}</loader-mock>;
    }
}