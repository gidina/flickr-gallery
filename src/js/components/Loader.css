.loader {
    /*border: 10px solid #f3f3f3;*/ /* Light grey */
    /*border-top: 10px solid #f55;*/
    border-radius: 50%;
    width: 50px;
    height: 50px;
    animation: spin 2s linear infinite;
    position: absolute;
    top: 40%;
    left: 45%;
}

@keyframes spin {
    0% { transform: rotate(0deg); }
    100% { transform: rotate(360deg); }
}