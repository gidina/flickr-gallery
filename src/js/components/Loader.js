import React, { Component } from 'react';
import './Loader.css';

class Loader extends Component {
  render() {
    const { color } = this.props;
    return (
        <div className="loader" style={{ border: '10px solid #f3f3f3', borderTop: `10px solid ${color}` }}></div>
    );
  }
}

export default Loader;