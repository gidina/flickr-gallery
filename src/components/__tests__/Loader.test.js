import React from "react";
import Loader from "../Loader";
import renderer from "react-test-renderer";

test("Renders correctly with pink color", () => {
    const color = "pink";
    const tree = renderer
        .create(<Loader color={color}/>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});