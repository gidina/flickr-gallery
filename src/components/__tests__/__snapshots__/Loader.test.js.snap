// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with pink color 1`] = `
<div
  className="loader"
  style={
    Object {
      "border": "10px solid #f3f3f3",
      "borderTop": "10px solid pink",
    }
  }
/>
`;
